### ISSUE TITLE
Request: connected underscores __

### ISSUE THREAD

**Issue by piskov** [kiliman/operator-mono-lig#87]
BEM web naming convention uses a lot of double underscores.

As of now Operator shows a little gap between two undercores. Would be nice to have it eliminated as in [7.2 Stylistic Sets Operator Mono](https://www.typography.com/fonts/operator/how-to-use#using_special_charactersl) (see “connected underscore”).

**Comment by kiliman** [comment]
That is actually built into the font.

![image](https://user-images.githubusercontent.com/47168/61139334-24725380-a497-11e9-9738-e18fa52ce3ae.png)

```html
<!DOCTYPE html>
<html>
  <head>
    <style>
      pre {
        font-family: 'Operator Mono SSm Lig';
      }
      .ss08 {
        font-feature-settings: 'ss08';
      }
    </style>
  </head>
  <body>
    <h2>Without Stylistic Set 08 Enabled</h2>
    <pre>
      .block__elem { color: #042; }
      .block__elem--modifier { color: #420; }
    </pre>
    <h2>With Stylistic Set 08 Enabled</h2>
    <pre class="ss08">
      .block__elem { color: #042; }
      .block__elem--modifier { color: #420; }
    </pre>
  </body>
</html>
```

The issue is that there isn't an easy to way to enable `font-feature-settings` directly in VS Code. There are some issues in GitHub asking for this feature.

https://github.com/Microsoft/vscode/issues/6918
https://github.com/Microsoft/vscode/issues/10435

The current workaround is to update the `workbench.main.css` file.

**Comment by piskov** [comment]
@kiliman could you please tell me what to add to worbench.main.css? I tried ↓ to no avail:
```.monaco-editor { font-feature-settings:"ss08"; }```

**Comment by kiliman** [comment]
I got it to work in VS Code. I also used the Custom CSS and Javascript Loader extension. https://github.com/be5invis/vscode-custom-css

This way I didn't need to modify the main css and can add a custom user file. Follow the instructions for the extension.

Make sure you add `!important`.

.monaco-editor {
    font-feature-settings: "ss08" !important;
}

Looks really nice
![image](https://user-images.githubusercontent.com/47168/61155121-1a178000-a4be-11e9-91df-1b3978afa0aa.png)

**Comment by piskov** [comment]
You’re messing with me right? :-)

Your code strips any stylistic sets from the original file. "ss08" only works with the original file but is pointless with the “lig” version.

Or is there any other version of your code (unmerged branch maybe)?

**Comment by kiliman** [comment]
Ah, that could be. Try the `features/contextsubst` branch. I rewrote the entire table generation to support contextual substitution. It only supports Operator Mono SSm Book (and part of OM Light) currently. That's why I haven't merged to master yet, since I need to redo all the ligatures :(

**Comment by piskov** [comment]
Yep, that has done the trick, thanks. But maybe you could make this a default behaviour withouth the need of switching to stylistic set? That would be nice.

To avoid installing any extension: find every mention of the `font-feature-settings:` in the `workbench.main.css` and add after a comma `"ss08"`. E. g.:
```
.monaco-editor {
  position:relative;
  overflow:visible;
  -webkit-text-size-adjust:100%;
  -webkit-font-feature-settings:"liga" off, "calt" off, "ss08";
  font-feature-settings:"liga" off, "calt" off, "ss08"
}

  .monaco-editor.enable-ligatures {
  -webkit-font-feature-settings:"liga" on, "calt" on, "ss08";
  font-feature-settings:"liga" on, "calt" on,  "ss08"
}
```

P. S. please keep up with the good work. I’ve reached out guys from Hoeffler. The response was, basically, no current plans for an update:
> We’ve had a few other people request these as well and while we don’t have any immediate plans to add them, it’s certainly on our radar.

**Comment by kiliman** [comment]
Added the __ ligature. Here's how it looks in VS Code. You no longer need to enable the stylistic set.

![image](https://user-images.githubusercontent.com/47168/61742961-4b1b6e80-ad62-11e9-8de3-6b150d42dc22.png)

**Comment by kiliman** [comment]
Closed by 95503e21c690ff369a5cb6d988ef6223f9e05549

**Comment by kiliman** [comment]
Please use the `develop` branch. I've merged the feature branch and deleted it.
